components/scheduler2/util.go

```diff
@@ -0,0 +1,57 @@
+package scheduler2
+
+import (
+	"fmt"
+	"time"
+
+	"go.temporal.io/server/service/history/consts"
+	"go.temporal.io/server/service/history/hsm"
+)
+
+// ValidateTask ensures that the given transition is possible with the current
+// state machine state.
+func ValidateTask[
+	S comparable,
+	SM hsm.StateMachine[S],
+	E any](node *hsm.Node, transition hsm.Transition[S, SM, E]) error {
+	if err := node.CheckRunning(); err != nil {
+		return err
+	}
+	sm, err := hsm.MachineData[SM](node)
+	if err != nil {
+		return err
+	}
+	if !transition.Possible(sm) {
+		return fmt.Errorf(
+			"%w: %w: cannot transition from state %v to %v",
+			consts.ErrStaleReference,
+			hsm.ErrInvalidTransition,
+			sm.State(),
+			transition.Destination,
+		)
+	}
+	return nil
+}
+
+// Generates a deterministic request ID for a buffered action's time. The request
```

Comment:
```suggestion
// GenerateRequestID generates a deterministic request ID for a buffered action's time. The request
```

Also consider not exporting this unless called outside of this package.

Reply:
Will fix. Most of these were public access from when I had them in separate packages, I'll give the package a full pass.